// baseui/es/side-navigation/side-navigation-next.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
import React, { createContext, isValidElement, useContext, useEffect, useState } from 'react';
import { useUID } from 'react-uid';
import { getOverrides } from '../helpers/overrides';
import ChevronDown from '../icon/chevron-down';
import ChevronUp from '../icon/chevron-up';
import { styled, useStyletron } from '../styles/index';
const SideNavigationContext = /*#__PURE__*/createContext({
  activeItemId: null
});
const StyledList = styled('ul', ({
  $open = true
}) => ({
  display: $open ? 'block' : 'none',
  margin: 0,
  padding: 0
}));
StyledList.displayName = "StyledList";
const StyledListItem = styled('li', {
  listStyleType: 'none'
});
StyledListItem.displayName = "StyledListItem";
export function SideNavigation(props) {
  const {
    activeItemId = null,
    children,
    overrides = {}
  } = props;
  const [List, listProps] = getOverrides(overrides.List, StyledList);
  return /*#__PURE__*/React.createElement(SideNavigationContext.Provider, {
    value: {
      activeItemId
    }
  }, /*#__PURE__*/React.createElement(List, listProps, children));
}
const StyledStartWrapper = styled('div', () => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-start'
}));
StyledStartWrapper.displayName = "StyledStartWrapper";
const StyledEndWrapper = styled('div', ({
  $theme
}) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-end',
  width: $theme.sizing.scale900
}));
StyledEndWrapper.displayName = "StyledEndWrapper";
const StyledStartEnhancerWrapper = styled('div', ({
  $theme
}) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'flex-start',
  width: $theme.sizing.scale950
}));
StyledStartEnhancerWrapper.displayName = "StyledStartEnhancerWrapper";
const StyledAction = styled('a', props => {
  const {
    $active,
    $clickable,
    $disabled,
    $indent,
    $theme
  } = props;
  const {
    colors,
    sizing,
    typography
  } = $theme;
  const paddingLeft = parseInt(sizing.scale600) + parseInt(sizing.scale950) * Math.max($indent, 0) + 'px';
  let style = {
    alignItems: 'center',
    background: 'none',
    borderTop: 'none',
    borderRight: 'none',
    borderBottom: 'none',
    borderLeft: $active ? `solid ${sizing.scale100} ${colors.contentPrimary}` : `solid ${sizing.scale100} transparent`,
    boxShadow: $active ? 'inset 999px 999px 0px rgb(0 0 0 / 8%)' : 'none',
    boxSizing: 'border-box',
    color: !$clickable || $active ? colors.contentPrimary : colors.contentSecondary,
    display: 'flex',
    font: 'inherit',
    justifyContent: 'space-between',
    overflowWrap: 'anywhere',
    paddingTop: sizing.scale550,
    paddingRight: sizing.scale600,
    paddingBottom: sizing.scale550,
    paddingLeft,
    textDecoration: 'none',
    width: '100%',
    ...typography.LabelMedium
  };

  if ($clickable) {
    style[':hover'] = {
      boxShadow: 'inset 999px 999px 0px rgb(0 0 0 / 4%)',
      cursor: 'pointer'
    };
  }

  if ($disabled) {
    style.color = colors.contentStateDisabled;
    style.cursor = 'not-allowed';
  }

  return style;
});
StyledAction.displayName = "StyledAction";

function Item(props) {
  const {
    ariaControls,
    ariaExpanded,
    active = false,
    disabled = false,
    endEnhancer,
    href,
    indent = 0,
    label,
    onClick,
    overrides = {},
    startEnhancer
  } = props;

  if (!label) {
    return null;
  }

  const [Action, actionProps] = getOverrides(overrides.Action, StyledAction);
  const [EndWrapper, endWrapperProps] = getOverrides(overrides.EndWrapper, StyledEndWrapper);
  const [StartEnhancerWrapper, startEnhancerWrapperProps] = getOverrides(overrides.StartEnhancerWrapper, StyledStartEnhancerWrapper);
  const [StartWrapper, startWrapperProps] = getOverrides(overrides.StartWrapper, StyledStartWrapper);
  const clickable = Boolean((onClick || href) && !disabled);
  const actionElement = clickable ? href ? 'a' : 'button' : 'div';
  return /*#__PURE__*/React.createElement(Action, _extends({
    $active: active,
    $as: actionElement,
    $clickable: clickable,
    $disabled: disabled,
    $indent: indent,
    "aria-controls": ariaControls,
    "aria-expanded": ariaExpanded,
    href: href,
    onClick: onClick,
    type: actionElement === 'button' ? 'button' : undefined
  }, actionProps), /*#__PURE__*/React.createElement(StartWrapper, startWrapperProps, startEnhancer && /*#__PURE__*/React.createElement(StartEnhancerWrapper, startEnhancerWrapperProps, startEnhancer), label), endEnhancer && /*#__PURE__*/React.createElement(EndWrapper, endWrapperProps, endEnhancer));
}

export function SideNavigationHeader({
  children,
  overrides: {
    ListItem,
    ...overrides
  } = {},
  ...props
}) {
  const [OverridedListItem, listItemProps] = getOverrides(ListItem, StyledListItem);
  return /*#__PURE__*/React.createElement(OverridedListItem, listItemProps, /*#__PURE__*/React.createElement(Item, _extends({}, props, {
    label: children,
    overrides: overrides
  })));
}
export function SideNavigationItem({
  children,
  id,
  overrides: {
    ListItem,
    ...overrides
  } = {},
  ...props
}) {
  const {
    activeItemId
  } = React.useContext(SideNavigationContext);
  const [OverridedListItem, listItemProps] = getOverrides(ListItem, StyledListItem);
  return /*#__PURE__*/React.createElement(OverridedListItem, listItemProps, /*#__PURE__*/React.createElement(Item, _extends({}, props, {
    label: children,
    active: activeItemId === id,
    overrides: overrides
  })));
}
export function SideNavigationSection(props) {
  const {
    active = false,
    children,
    forceOpen = false,
    indent = 0,
    overrides: {
      List,
      ListItem,
      ...overrides
    } = {},
    startEnhancer,
    title
  } = props;
  const [, theme] = useStyletron();
  const {
    activeItemId
  } = useContext(SideNavigationContext);
  const [open, setOpen] = useState(false);
  const uid = useUID();
  useEffect(() => {
    let childIsActive = false;

    if (activeItemId != null) {
      let stack = React.Children.toArray(children);

      while (stack.length) {
        const current = stack.pop();

        if ( /*#__PURE__*/isValidElement(current)) {
          if (current.props.id === activeItemId) {
            childIsActive = true;
            break;
          }

          if (current.props.children) {
            stack = stack.concat(React.Children.toArray(current.props.children));
          }
        }
      }
    }

    if (childIsActive) {
      setOpen(true);
    } else {
      setOpen(false);
    }
  }, [activeItemId, children]);
  useEffect(() => {
    if (forceOpen) {
      setOpen(true);
    }
  }, [forceOpen]);

  if (React.Children.toArray(children).filter(React.isValidElement).length === 0) {
    return null;
  }

  const Chevron = open ? ChevronUp : ChevronDown;
  const [OverridedList, overridedListProps] = getOverrides(List, StyledList);
  const [OverridedListItem, overridedListItemProps] = getOverrides(ListItem, StyledListItem);
  return /*#__PURE__*/React.createElement(OverridedListItem, overridedListItemProps, /*#__PURE__*/React.createElement(Item, {
    ariaControls: uid,
    ariaExpanded: open,
    endEnhancer: /*#__PURE__*/React.createElement(Chevron, {
      title: "",
      size: theme.sizing.scale800
    }),
    label: title,
    indent: indent,
    onClick: () => setOpen(prev => !prev),
    overrides: overrides,
    startEnhancer: startEnhancer
  }), /*#__PURE__*/React.createElement(OverridedList, _extends({
    id: uid,
    $open: open
  }, overridedListProps), React.Children.map(children, child => child === null ? null : /*#__PURE__*/React.cloneElement(child, {
    indent: indent + 1
  }))));
}
