// baseui/esm/system-banner/system-banner.js
var _excluded = ["artworkIcon", "primaryAction", "secondaryAction", "overrides"],
    _excluded2 = ["Root", "LeadingContent", "TrailingIconButton", "MessageContent"];

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
import * as React from 'react';
import { Banner, ARTWORK_TYPE, HIERARCHY } from '../banner';
import { StyledRoot, StyledLeadingContent, StyledTrailingIconButton, StyledMessageContent } from './styled-components';
import { mergeOverride } from '../helpers/overrides';
export function SystemBanner(props) {
  var artworkIcon = props.artworkIcon,
      primaryAction = props.primaryAction,
      secondaryAction = props.secondaryAction,
      _props$overrides = props.overrides,
      overrides = _props$overrides === void 0 ? {} : _props$overrides,
      restProps = _objectWithoutProperties(props, _excluded);

  var Root = overrides.Root,
      LeadingContent = overrides.LeadingContent,
      TrailingIconButton = overrides.TrailingIconButton,
      MessageContent = overrides.MessageContent,
      restOverrides = _objectWithoutProperties(overrides, _excluded2);

  var hasSinglePrimaryAction = (primaryAction === null || primaryAction === void 0 ? void 0 : primaryAction.onClick) && !secondaryAction;
  var hasTwoActions = (primaryAction === null || primaryAction === void 0 ? void 0 : primaryAction.onClick) && (secondaryAction === null || secondaryAction === void 0 ? void 0 : secondaryAction.onClick);
  var hasTrailingIcon = hasSinglePrimaryAction && primaryAction.icon;
  var systemBannerRootOverrides = {
    component: StyledRoot,
    props: hasSinglePrimaryAction ? {
      $isButton: true,
      $as: 'button',
      onClick: primaryAction.onClick,
      'aria-label': primaryAction.label
    } : {}
  };
  var systemBannerLeadingContentOverrides = {
    component: StyledLeadingContent
  };
  var styledTrailingIconButtonOverrides = {
    component: StyledTrailingIconButton,
    props: {
      $as: 'div'
    }
  };
  var systemBannerMessageContentOverrides = {
    component: StyledMessageContent,
    props: {
      $as: 'button',
      onClick: primaryAction === null || primaryAction === void 0 ? void 0 : primaryAction.onClick,
      'aria-label': primaryAction === null || primaryAction === void 0 ? void 0 : primaryAction.label
    }
  };

  var systemBannerOverrides = _objectSpread({
    Root: mergeOverride(systemBannerRootOverrides, Root || {}),
    LeadingContent: mergeOverride(systemBannerLeadingContentOverrides, LeadingContent || {}),
    TrailingIconButton: hasTrailingIcon ? mergeOverride(styledTrailingIconButtonOverrides, TrailingIconButton || {}) : {},
    MessageContent: hasTwoActions ? mergeOverride(systemBannerMessageContentOverrides, MessageContent || {}) : {}
  }, restOverrides);

  return /*#__PURE__*/React.createElement(Banner, _extends({
    action: secondaryAction ? _objectSpread({
      onClick: null
    }, secondaryAction) : hasTrailingIcon ? {
      icon: primaryAction.icon,
      onClick: null,
      label: null
    } : null,
    hierarchy: HIERARCHY.high,
    overrides: systemBannerOverrides
  }, artworkIcon ? {
    artwork: {
      icon: artworkIcon,
      type: ARTWORK_TYPE.icon
    }
  } : {}, restProps));
}
