// baseui/esm/sheet/sheet.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
import * as React from 'react';
import { getOverrides } from '../helpers/overrides';
import { ProgressBar, SIZE } from '../progress-bar';
import { SIZE as BTN_SIZE } from '../button';
import { SIZE as DIVIDER_SIZE } from '../divider';
import { ActionButton as DefaultActionButton } from './action-button';
import { StyledBottomContainer, StyledBody, StyledDescription, StyledDivider, StyledEmptyDiv, StyledGrabber, StyledHeader, StyledHeaderGrid, StyledHeaderTextContainer, StyledRoot, StyledTitle, StyledTopContainer } from './styled-components';

function renderBodyContent(content) {
  return typeof content === 'function' ? content() : content;
}

function renderActionButtonContent(content, buttonSize) {
  if (typeof content === 'string') {
    return content;
  }

  var Icon = content;
  return /*#__PURE__*/React.createElement(Icon, {
    size: buttonSize === BTN_SIZE.compact ? 22 : 26
  });
}

export var Sheet = function Sheet(_ref) {
  var children = _ref.children,
      content = _ref.content,
      description = _ref.description,
      _ref$draggable = _ref.draggable,
      draggable = _ref$draggable === void 0 ? false : _ref$draggable,
      leadingAction = _ref.leadingAction,
      _ref$overrides = _ref.overrides,
      overrides = _ref$overrides === void 0 ? {} : _ref$overrides,
      progressBar = _ref.progressBar,
      title = _ref.title,
      topPosition = _ref.topPosition,
      trailingAction = _ref.trailingAction;

  var _getOverrides = getOverrides(overrides.Root, StyledRoot),
      _getOverrides2 = _slicedToArray(_getOverrides, 2),
      Root = _getOverrides2[0],
      rootProps = _getOverrides2[1];

  var _getOverrides3 = getOverrides(overrides.TopContainer, StyledTopContainer),
      _getOverrides4 = _slicedToArray(_getOverrides3, 2),
      TopContainer = _getOverrides4[0],
      topContainerProps = _getOverrides4[1];

  var _getOverrides5 = getOverrides(overrides.BottomContainer, StyledBottomContainer),
      _getOverrides6 = _slicedToArray(_getOverrides5, 2),
      BottomContainer = _getOverrides6[0],
      bottomContainerProps = _getOverrides6[1];

  var _getOverrides7 = getOverrides(overrides.Header, StyledHeader),
      _getOverrides8 = _slicedToArray(_getOverrides7, 2),
      Header = _getOverrides8[0],
      headerProps = _getOverrides8[1];

  var _getOverrides9 = getOverrides(overrides.Grabber, StyledGrabber),
      _getOverrides10 = _slicedToArray(_getOverrides9, 2),
      Grabber = _getOverrides10[0],
      grabberProps = _getOverrides10[1];

  var _getOverrides11 = getOverrides(overrides.HeaderGrid, StyledHeaderGrid),
      _getOverrides12 = _slicedToArray(_getOverrides11, 2),
      HeaderGrid = _getOverrides12[0],
      headerGridProps = _getOverrides12[1];

  var _getOverrides13 = getOverrides(overrides.ActionButton, DefaultActionButton),
      _getOverrides14 = _slicedToArray(_getOverrides13, 2),
      ActionButton = _getOverrides14[0],
      actionButtonProps = _getOverrides14[1];

  var _getOverrides15 = getOverrides(overrides.EmptyDiv, StyledEmptyDiv),
      _getOverrides16 = _slicedToArray(_getOverrides15, 2),
      EmptyDiv = _getOverrides16[0],
      emptyDivProps = _getOverrides16[1];

  var _getOverrides17 = getOverrides(overrides.HeaderTextContainer, StyledHeaderTextContainer),
      _getOverrides18 = _slicedToArray(_getOverrides17, 2),
      HeaderTextContainer = _getOverrides18[0],
      HeaderTextContainerProps = _getOverrides18[1];

  var _getOverrides19 = getOverrides(overrides.Title, StyledTitle),
      _getOverrides20 = _slicedToArray(_getOverrides19, 2),
      Title = _getOverrides20[0],
      titleProps = _getOverrides20[1];

  var _getOverrides21 = getOverrides(overrides.Description, StyledDescription),
      _getOverrides22 = _slicedToArray(_getOverrides21, 2),
      Description = _getOverrides22[0],
      descriptionProps = _getOverrides22[1];

  var _getOverrides23 = getOverrides(overrides.Divider, StyledDivider),
      _getOverrides24 = _slicedToArray(_getOverrides23, 2),
      Divider = _getOverrides24[0],
      dividerProps = _getOverrides24[1];

  var _getOverrides25 = getOverrides(overrides.Body, StyledBody),
      _getOverrides26 = _slicedToArray(_getOverrides25, 2),
      Body = _getOverrides26[0],
      bodyProps = _getOverrides26[1];

  var hasTitle = Boolean(title);
  var hasDescription = Boolean(description);
  var hasLeadingAction = Boolean(leadingAction);
  var hasTrailingAction = Boolean(trailingAction);
  var buttonSize = hasTitle ? BTN_SIZE.default : BTN_SIZE.compact;
  var shouldDisplayHeader = draggable || hasTitle || hasDescription || hasLeadingAction || hasTrailingAction;
  return /*#__PURE__*/React.createElement(Root, _extends({
    $draggable: draggable
  }, rootProps), /*#__PURE__*/React.createElement(TopContainer, _extends({
    $draggable: draggable,
    $topPosition: topPosition
  }, topContainerProps), children), /*#__PURE__*/React.createElement(BottomContainer, _extends({
    $draggable: draggable,
    $topPosition: topPosition
  }, bottomContainerProps), shouldDisplayHeader && /*#__PURE__*/React.createElement(Header, headerProps, draggable && /*#__PURE__*/React.createElement(Grabber, grabberProps), /*#__PURE__*/React.createElement(HeaderGrid, _extends({
    $hasLeadingAction: hasLeadingAction,
    $hasTrailingAction: hasTrailingAction
  }, headerGridProps), leadingAction && /*#__PURE__*/React.createElement(ActionButton, _extends({
    onClick: leadingAction.onClick,
    "aria-label": leadingAction.label,
    size: buttonSize
  }, actionButtonProps), renderActionButtonContent(leadingAction.renderIcon || leadingAction.label, buttonSize)), trailingAction && !leadingAction && /*#__PURE__*/React.createElement(EmptyDiv, emptyDivProps), (hasTitle || hasDescription) && /*#__PURE__*/React.createElement(HeaderTextContainer, _extends({
    $draggable: draggable,
    $hasTitle: hasTitle,
    $hasDescription: hasDescription
  }, HeaderTextContainerProps), title && /*#__PURE__*/React.createElement(Title, titleProps, title), description && /*#__PURE__*/React.createElement(Description, descriptionProps, description)), trailingAction && /*#__PURE__*/React.createElement(ActionButton, _extends({
    onClick: trailingAction.onClick,
    "aria-label": trailingAction.label,
    size: buttonSize
  }, actionButtonProps), renderActionButtonContent(trailingAction.renderIcon || trailingAction.label, buttonSize)), !trailingAction && leadingAction && /*#__PURE__*/React.createElement(EmptyDiv, emptyDivProps)), progressBar ? /*#__PURE__*/React.createElement(ProgressBar, _extends({
    size: SIZE.small,
    overrides: {
      BarContainer: {
        style: {
          marginTop: 0,
          marginBottom: 0
        }
      }
    }
  }, progressBar)) : /*#__PURE__*/React.createElement(Divider, _extends({
    $size: DIVIDER_SIZE.section
  }, dividerProps))), /*#__PURE__*/React.createElement(Body, bodyProps, renderBodyContent(content))));
};
export default Sheet;
