// baseui/system-banner/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledTrailingIconButton = exports.StyledRoot = exports.StyledMessageContent = exports.StyledLeadingContent = void 0;

var _styles = require("../styles");

var _banner = require("../banner");

var _responsiveHelpers = require("../helpers/responsive-helpers");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var StyledRoot = (0, _styles.withStyle)(_banner.StyledRoot, function (_ref) {
  var $isButton = _ref.$isButton,
      $theme = _ref.$theme;
  return _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, $isButton ? {
    ':hover': {
      boxShadow: 'inset 999px 999px 0px rgba(0, 0, 0, 0.04)'
    },
    ':active': {
      boxShadow: 'inset 999px 999px 0px rgba(0, 0, 0, 0.08)'
    },
    // button style reset
    '-webkit-appearance': 'none',
    textAlign: 'inherit',
    boxShadow: 'none',
    cursor: 'pointer',
    border: 'none',
    alignItems: 'center'
  } : {}), (0, _responsiveHelpers.getMediaQueryPageMargins)($theme)), (0, _responsiveHelpers.getMinimumPageMargins)($theme.grid.margins)), {}, {
    borderRadius: 0,
    margin: 0
  });
});
exports.StyledRoot = StyledRoot;
StyledRoot.displayName = "StyledRoot";
StyledRoot.displayName = 'StyledRoot';
var StyledLeadingContent = (0, _styles.withStyle)(_banner.StyledLeadingContent, function () {
  return {
    padding: 0
  };
});
exports.StyledLeadingContent = StyledLeadingContent;
StyledLeadingContent.displayName = "StyledLeadingContent";
StyledLeadingContent.displayName = 'StyledLeadingContent';
var StyledTrailingIconButton = (0, _styles.withStyle)(_banner.StyledTrailingIconButton, function () {
  return {
    ':hover': {
      boxShadow: 'none'
    },
    ':active': {
      boxShadow: 'none'
    }
  };
});
exports.StyledTrailingIconButton = StyledTrailingIconButton;
StyledTrailingIconButton.displayName = "StyledTrailingIconButton";
StyledTrailingIconButton.displayName = 'StyledTrailingIconButton';
var StyledMessageContent = (0, _styles.withStyle)(_banner.StyledMessageContent, function () {
  return {
    // button styles reset
    backgroundColor: 'transparent',
    borderBottomWidth: 0,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    borderTopWidth: 0,
    fontFamily: 'inherit',
    fontSize: 'inherit',
    fontStyle: 'inherit',
    fontWeight: 'inherit',
    lineHeight: 'inherit',
    cursor: 'pointer',
    ':hover': {
      boxShadow: 'inset 999px 999px 0px rgba(0, 0, 0, 0.04)'
    },
    ':active': {
      boxShadow: 'inset 999px 999px 0px rgba(0, 0, 0, 0.08)'
    }
  };
});
exports.StyledMessageContent = StyledMessageContent;
StyledMessageContent.displayName = "StyledMessageContent";
StyledMessageContent.displayName = 'StyledMessageContent';
